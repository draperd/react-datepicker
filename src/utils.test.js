// @flow

// import React from "react";
import {
  datesAreEqual,
  getFirstDateInMonth,
  getFirstDateOfWeek,
  getLastDateInMonth,
  getMonthData,
  getNextDay,
  getWeekData
} from "./utils";
// import { act } from "react-dom/test-utils";
// import { mount } from "enzyme";
// import chai, { expect as chaiExpect } from "chai";
// import chaiEnzyme from "chai-enzyme";

// chai.use(chaiEnzyme());

describe("datesAreEqual", () => {
  const date1 = new Date("2020-03-04");
  const date2 = new Date("2018-02-15");
  it("should return true when dates are equal", () => {
    expect(datesAreEqual({ date1, date2: date1 })).toBe(true);
  });

  it("should return false when dates are NOT equal", () => {
    expect(datesAreEqual({ date1, date2 })).toBe(false);
  });
});

describe("getFirstDateOfWeek", () => {
  it("should return correct date for Feb 2020", () => {
    const date = new Date("2020-02-15");
    const firstDateOfWeek = getFirstDateOfWeek({ date });
    expect(firstDateOfWeek.toDateString()).toBe("Sun Feb 09 2020");
  });

  it("should return correct date for Nov 2019", () => {
    const date = new Date("2019-11-13");
    const firstDateOfWeek = getFirstDateOfWeek({ date });
    expect(firstDateOfWeek.toDateString()).toBe("Sun Nov 10 2019");
  });

  it("should return correct date for Jan 2019", () => {
    const date = new Date("2019-01-02");
    const firstDateOfWeek = getFirstDateOfWeek({ date });
    expect(firstDateOfWeek.toDateString()).toBe("Sun Dec 30 2018");
  });
});

describe("getFirstDateInMonth", () => {
  it("should return first date", () => {
    const date = new Date("2020-02-15");
    const firstDateInMonth = getFirstDateInMonth({ date });
    expect(firstDateInMonth.toDateString()).toBe("Sat Feb 01 2020");
  });
});

describe("getLastDateInMonth", () => {
  it("should return last date", () => {
    const date = new Date("2020-02-15");
    const lastDateInMonth = getLastDateInMonth({ date });
    expect(lastDateInMonth.toDateString()).toBe("Sat Feb 29 2020");
  });
});

describe("getNextDay", () => {
  it("adds a day in same month", () => {
    const date = new Date("2019-01-02");
    const firstDateOfWeek = getNextDay({ date });
    expect(firstDateOfWeek.toDateString()).toBe("Thu Jan 03 2019");
  });

  it("adds a month and year for last day of year", () => {
    const date = new Date("2019-12-31");
    const firstDateOfWeek = getNextDay({ date });
    expect(firstDateOfWeek.toDateString()).toBe("Wed Jan 01 2020");
  });
});

describe("getWeekData", () => {
  it("gets correct data for week", () => {
    const date = new Date("2019-01-02");
    const firstDateInMonth = getFirstDateOfWeek({ date });
    const lastDateInMonth = getLastDateInMonth({ date });
    const today = new Date();
    const weekData = getWeekData({
      date,
      firstDateInMonth,
      lastDateInMonth,
      selectedDate: date,
      today
    });
    expect(weekData).toMatchSnapshot();
  });
});

describe("getMonthData", () => {
  it("gets correct data for month", () => {
    const date = new Date("2019-01-02");
    const today = new Date();
    const monthData = getMonthData({ date, today });
    expect(monthData).toMatchSnapshot();
  });
});
