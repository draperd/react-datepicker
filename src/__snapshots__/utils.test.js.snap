// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMonthData gets correct data for month 1`] = `
Array [
  Object {
    "0": Object {
      "available": true,
      "date": 2018-12-29T13:00:00.000Z,
      "dayOfMonth": 30,
      "isInCurrentMonth": false,
      "selected": false,
      "today": false,
    },
    "1": Object {
      "available": true,
      "date": 2018-12-30T13:00:00.000Z,
      "dayOfMonth": 31,
      "isInCurrentMonth": false,
      "selected": false,
      "today": false,
    },
    "2": Object {
      "available": true,
      "date": 2018-12-31T13:00:00.000Z,
      "dayOfMonth": 1,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "3": Object {
      "available": true,
      "date": 2019-01-01T13:00:00.000Z,
      "dayOfMonth": 2,
      "isInCurrentMonth": true,
      "selected": true,
      "today": false,
    },
    "4": Object {
      "available": true,
      "date": 2019-01-02T13:00:00.000Z,
      "dayOfMonth": 3,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "5": Object {
      "available": true,
      "date": 2019-01-03T13:00:00.000Z,
      "dayOfMonth": 4,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "6": Object {
      "available": true,
      "date": 2019-01-04T13:00:00.000Z,
      "dayOfMonth": 5,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
  },
  Object {
    "0": Object {
      "available": true,
      "date": 2019-01-05T13:00:00.000Z,
      "dayOfMonth": 6,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "1": Object {
      "available": true,
      "date": 2019-01-06T13:00:00.000Z,
      "dayOfMonth": 7,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "2": Object {
      "available": true,
      "date": 2019-01-07T13:00:00.000Z,
      "dayOfMonth": 8,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "3": Object {
      "available": true,
      "date": 2019-01-08T13:00:00.000Z,
      "dayOfMonth": 9,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "4": Object {
      "available": true,
      "date": 2019-01-09T13:00:00.000Z,
      "dayOfMonth": 10,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "5": Object {
      "available": true,
      "date": 2019-01-10T13:00:00.000Z,
      "dayOfMonth": 11,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "6": Object {
      "available": true,
      "date": 2019-01-11T13:00:00.000Z,
      "dayOfMonth": 12,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
  },
  Object {
    "0": Object {
      "available": true,
      "date": 2019-01-12T13:00:00.000Z,
      "dayOfMonth": 13,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "1": Object {
      "available": true,
      "date": 2019-01-13T13:00:00.000Z,
      "dayOfMonth": 14,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "2": Object {
      "available": true,
      "date": 2019-01-14T13:00:00.000Z,
      "dayOfMonth": 15,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "3": Object {
      "available": true,
      "date": 2019-01-15T13:00:00.000Z,
      "dayOfMonth": 16,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "4": Object {
      "available": true,
      "date": 2019-01-16T13:00:00.000Z,
      "dayOfMonth": 17,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "5": Object {
      "available": true,
      "date": 2019-01-17T13:00:00.000Z,
      "dayOfMonth": 18,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "6": Object {
      "available": true,
      "date": 2019-01-18T13:00:00.000Z,
      "dayOfMonth": 19,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
  },
  Object {
    "0": Object {
      "available": true,
      "date": 2019-01-19T13:00:00.000Z,
      "dayOfMonth": 20,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "1": Object {
      "available": true,
      "date": 2019-01-20T13:00:00.000Z,
      "dayOfMonth": 21,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "2": Object {
      "available": true,
      "date": 2019-01-21T13:00:00.000Z,
      "dayOfMonth": 22,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "3": Object {
      "available": true,
      "date": 2019-01-22T13:00:00.000Z,
      "dayOfMonth": 23,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "4": Object {
      "available": true,
      "date": 2019-01-23T13:00:00.000Z,
      "dayOfMonth": 24,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "5": Object {
      "available": true,
      "date": 2019-01-24T13:00:00.000Z,
      "dayOfMonth": 25,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "6": Object {
      "available": true,
      "date": 2019-01-25T13:00:00.000Z,
      "dayOfMonth": 26,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
  },
  Object {
    "0": Object {
      "available": true,
      "date": 2019-01-26T13:00:00.000Z,
      "dayOfMonth": 27,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "1": Object {
      "available": true,
      "date": 2019-01-27T13:00:00.000Z,
      "dayOfMonth": 28,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "2": Object {
      "available": true,
      "date": 2019-01-28T13:00:00.000Z,
      "dayOfMonth": 29,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "3": Object {
      "available": true,
      "date": 2019-01-29T13:00:00.000Z,
      "dayOfMonth": 30,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "4": Object {
      "available": true,
      "date": 2019-01-30T13:00:00.000Z,
      "dayOfMonth": 31,
      "isInCurrentMonth": true,
      "selected": false,
      "today": false,
    },
    "5": Object {
      "available": true,
      "date": 2019-01-31T13:00:00.000Z,
      "dayOfMonth": 1,
      "isInCurrentMonth": false,
      "selected": false,
      "today": false,
    },
    "6": Object {
      "available": true,
      "date": 2019-02-01T13:00:00.000Z,
      "dayOfMonth": 2,
      "isInCurrentMonth": false,
      "selected": false,
      "today": false,
    },
  },
]
`;

exports[`getWeekData gets correct data for week 1`] = `
Object {
  "0": Object {
    "available": true,
    "date": 2018-12-30T00:00:00.000Z,
    "dayOfMonth": 30,
    "isInCurrentMonth": true,
    "selected": false,
    "today": false,
  },
  "1": Object {
    "available": true,
    "date": 2018-12-31T00:00:00.000Z,
    "dayOfMonth": 31,
    "isInCurrentMonth": true,
    "selected": false,
    "today": false,
  },
  "2": Object {
    "available": true,
    "date": 2019-01-01T00:00:00.000Z,
    "dayOfMonth": 1,
    "isInCurrentMonth": true,
    "selected": false,
    "today": false,
  },
  "3": Object {
    "available": true,
    "date": 2019-01-02T00:00:00.000Z,
    "dayOfMonth": 2,
    "isInCurrentMonth": true,
    "selected": true,
    "today": false,
  },
  "4": Object {
    "available": true,
    "date": 2019-01-03T00:00:00.000Z,
    "dayOfMonth": 3,
    "isInCurrentMonth": true,
    "selected": false,
    "today": false,
  },
  "5": Object {
    "available": true,
    "date": 2019-01-04T00:00:00.000Z,
    "dayOfMonth": 4,
    "isInCurrentMonth": true,
    "selected": false,
    "today": false,
  },
  "6": Object {
    "available": true,
    "date": 2019-01-05T00:00:00.000Z,
    "dayOfMonth": 5,
    "isInCurrentMonth": true,
    "selected": false,
    "today": false,
  },
}
`;
